src/backend/core/api/permissions.py

```diff
@@ -59,6 +62,41 @@ def has_object_permission(self, request, view, obj):
             return False
 
 
+class CanManageInvitationPermission(permissions.BasePermission):
+    """
+    Custom permission class to handle permission checks for managing invitations.
+    """
+
+    def has_permission(self, request, view):
+        user = request.user
+
+        # Ensure the user is authenticated
+        if not (bool(request.auth) or request.user.is_authenticated):
+            return False
+
+        # Apply permission checks only for creation (POST requests)
+        if request.method != "POST":
```

Comment:
In fact, in the case of update, I have an invitation ID and I can rely on the `has_object_permission` method to determine access. I don't need to make these checks here and it would just be one more db query before calling the `has_object_permission` method and doing redundant checks right?

Reply:
Alright, then the name of the permission is misleading, it should be "create" instead of "manage", what do you think?

Reply:
And maybe check the "create" action instead of the POST method? (Not sure about this comment ^^, but you may have an opinion on this)

Reply:
It seems to be equivalent: https://github.com/encode/django-rest-framework/blob/d3dd45b3f48856c3513ab1eb5354194fa9898f39/rest_framework/routers.py#L101

I can opt for the create action as it is more explicit